subsys/dfu/src/dfu_target_mcuboot.c

```diff
@@ -75,6 +109,31 @@ int dfu_target_mcuboot_init(size_t file_size)
 {
 	int err = flash_img_init(&flash_img);
 
+	if (IS_ENABLED(CONFIG_SETTINGS_FCB)) {
+		err = settings_subsys_init();
+		static struct settings_handler sh = {
+			.name = MODULE,
+			.h_set = settings_set,
+		};
+		if (err) {
+			LOG_ERR("settings_subsys_init failed (err %d)", err);
+			return err;
+		}
+
+		err = settings_register(&sh);
+
+		if (err) {
+			LOG_ERR("Cannot register settings (err %d)", err);
+			return err;
+		}
+
+	}
+
+	if (err) {
```

Comment:
This check needs to be rewritten. There is no error check for the `err` returned from `flash_img_init`.

I'm not sure what this `err` is supposed to be, but to me it looks like the comment in this `LOG_WRN` should be updated and the check be moved directly below the `flash_img_init()`.

Reply:
I see now that the `err` check for `flash_img_init` is intact further down, this should be moved up.

Reply:
Done